### ISSUE TITLE
how to remove the node?

### ISSUE THREAD

**Issue by asyncins** [causal-agent/scraper#125]
I want remove the `p` node, example:
```
use scraper::{Html, Selector};

fn main() {
    let html = r#"
        <html>
            <head>
                <title>Example News Article</title>
            </head>
            <body>
                <h1>Example News Article</h1>
                <div class="article">
                    <h2>Article 1</h2>
                    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
                </div>
                <div class="article">
                    <h2>Article 2</h2>
                    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
                </div>
            </body>
        </html>
    "#;

    let document = Html::parse_document(html);

    let p_selector = Selector::parse("p").unwrap();
    for p_node in document.select(&p_selector) {
        p_node.remove_from_parent();
    }
    println!("{}", document.root_element().html());
}

```

=========

the error: no method named `remove_from_parent` found for struct `ElementRef<'_>` in the current scope

What do you propose ???

**Comment by adamreichold** [comment]
I think the method you are looking for is called [`detach`](https://docs.rs/ego-tree/0.6.2/ego_tree/struct.NodeMut.html#method.detach).

In any case, you cannot modify the tree while you iterating it (through the node references, via the selector), so I would suggest collecting the node identifiers and detaching them in a second pass:

```rust
let p_selector = Selector::parse("p").unwrap();
let ids = document.select(&p_selector).map(|p_node| p_node.id()).collect::<Vec<_>>();

for id in ids {
    document.tree.get_mut(id).unwrap().detach();
}
```
